packages/ses/src/module-instance.js

```diff
@@ -73,6 +73,41 @@ export const makeThirdPartyModuleInstance = (
         update(value);
       };
     });
+    // Default was not in exports, need an implicit default
+    if (!notifiers.default) {
```

Comment:
```suggestion
    // For third-party module records that do not export a `default` name,
    // we infer that the default export is the namespace object itself.
    if (!notifiers.default) {
```

As I write this comment, I wonder whether it’s better or possible to push this concern to the CommonJS module record, forcing it to provide a `default` name, but only if there’s no explicit `default` export. I believe it’s valid for a module to not have an exported variable named `default`.

I suppose the litmus test for whether to include this special behavior here would be: is this implementation equally valid for WASM? Could it possibly interfere with intentional behavior of a module? Could it encourage dependees to depend on the non-existence of an export named `default`, and if they did, would that make their code more accidentally fragile? Is adding a `default` export to a WASM or CommonJS module a breaking change?

Reply:
In import from cjs running prettier can be a breaking change thanks to lexer. I get the concern and I think the ecosystem is going to deal with it in certain ways we'll need to accommodate. When I posted about it people brought up " the infamous triplet" which makes sure the consumer gets the right reference no matter how they import it. It makes a mess, adds a default field visible in cjs exports and so on.

Moving default adding to parse-cjs is problematic because to make it wire up correctly of have to declare it early on because at execution time it's already to late. And that would make the synthetic default appear in cjs require cjs too. At least my mental model says so.

The way esm cjs interop works AFAIR there's no way to not have a default so adding an explicit default +/- __esModule is something to dive into via the compat table.

Looking at e2e tests this seems to be working fine most of the time. I think I saw one that maybe produces an unexpected default where it wasn't supposed to be, I'll check it more thoroughly.  


Having said that, I'll happily experiment with wasm to see if this has potential to interfere.
Any prior at on loading wasm in Endo?

Reply:
Ok, so I implemented wasm support in compartment mapper and the implicit default does show up unnecessarily. 
Let's see if I can fix it today.